src/main/java/net/neoforged/art/internal/ClassProviderImpl.java

```diff
@@ -179,9 +179,25 @@ public Collection<? extends IFieldInfo> getFields() {
             return fieldsView;
         }
 
+        @Deprecated
         @Override
         public Optional<? extends IFieldInfo> getField(String name) {
-            return fields == null ? Optional.empty() : Optional.ofNullable(this.fields.get(name));
+            if (this.fields != null) {
+                for (Map.Entry<String, FieldInfo> entry : this.fields.entrySet()) {
+                    if (entry.getKey().split(" ")[0].equals(name)) {
+                        // Backwards compatibility not possible when duplication
+                        // on field names, constructor was crashing without descriptor logic
+                        // Let it be any in this case
+                        return Optional.of(entry.getValue());
+                    }
+                }
+            }
+            return Optional.empty();
+        }
+
+        @Override
+        public Optional<? extends IFieldInfo> getField(String name, String desc) {
+            return this.fields == null ? Optional.empty() : Optional.ofNullable(this.fields.get(name + " " + desc));
```

Comment:
Spaces are valid in identifiers, please use `:` instead,

Reply:
`:` is also value in field names

Reply:
A good idea would be to use `.` here -- it _is_ a reserved character that can't be in field names.